src/main/java/com/solvd/eshop/page/HomePage.java

```diff
@@ -3,50 +3,55 @@
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
-import org.openqa.selenium.support.PageFactory;
-import org.openqa.selenium.support.ui.ExpectedConditions;
-import org.openqa.selenium.support.ui.WebDriverWait;
 
-import java.time.Duration;
+public class HomePage extends AbstractPage{
 
-public class HomePage {
-        protected WebDriver driver;
-        private WebDriverWait waiter;
+    @FindBy(css = ".styles_reactButton__2olKd")
+    private WebElement cookieButton;
 
-        @FindBy(css = ".styles_reactButton__2olKd")
-        private WebElement cookieButton;
+    @FindBy(css = ".styles_userToolsToggler__imcSl")
+    private WebElement accountMenu;
 
-        @FindBy(css = ".styles_userToolsToggler__imcSl")
-        private WebElement accountMenu;
+    @FindBy(css = ".userToolsBtn")
+    private WebElement loginButton;
 
-        @FindBy(css = ".userToolsBtn")
-        private WebElement loginButton;
+    @FindBy (css = ".styles_userTools__2J7cp.undefined")
+    private WebElement accountMenuAfterLogin;
 
+    @FindBy(css = ".userToolsSubtitle")
+    private WebElement userToolsTitle;
 
-        public HomePage(WebDriver driver) {
-            this.driver = driver;
-            this.waiter = new WebDriverWait(this.driver,Duration.ofMillis(1000));
-            PageFactory.initElements(driver, this);
+    public HomePage(WebDriver driver) {
+        super(driver);
+    }
+
+    public void clickCookieButton() {
+        if (cookieButton.isDisplayed()) {
+            elementClick(cookieButton,0);
         }
+    }
 
-        public void clickCookieButton() {
-            if (cookieButton.isDisplayed()) {
-                cookieButton.click();
-            }
+    public void clickAccountMenu() {
+        if (accountMenu.isDisplayed()) {
+            elementClick(accountMenu,0);
         }
+    }
 
-        public void clickAccountMenu() {
-            if (accountMenu.isDisplayed()) {
-                accountMenu.click();
-            }
+    public LoginPage clickLoginButton() {
+        if (loginButton.isDisplayed()) {
+            elementClick(loginButton,5);
         }
+        return new LoginPage(driver);
+    }
 
-        public LoginPage clickLoginButton() {
-            LoginPage loginPage = new LoginPage(driver);
-            waiter.until(ExpectedConditions.elementToBeClickable(loginButton));
-            if (loginButton.isDisplayed()) {
-                loginButton.click();
-            }
-            return loginPage;
+    public void clickAccountMenuAfterLogin() {
+        if (accountMenuAfterLogin.isDisplayed()) {
+            elementClick(accountMenuAfterLogin, 5);
         }
+    }
```

Comment:
If you add **if (someElement.isDisplayed())** before click on this element, it'll be difficult to understand why test failed. For me better variant to move this isDisplayed to **elementClick** method and add some exception or at least logger with message that element isn't visible.

Reply:
someElement.isDisplayed()) was moved to elementClick method and logger was added